### PR TITLE
1.Change timezone

### PR DESCRIPTION
**Pull request by maiklgravi**

### CODE REVIEW

File: app/Models/Film.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    use HasFactory;
+    public function videoname()
+    {
+        return $this->hasMany(FilmVideoName::class);
+    }
+    public function alternativname()
+    {
+        return $this->hasMany(FilmAlternativName::class);
+    }
+    public function gender()
```
Comment (vistrati) [inline app/Models/Film.php:19]: I think genre is more suitable name for this method.

File: app/Models/Film.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Film extends Model
+{
+    use HasFactory;
+    public function videoname()
```
Comment (vistrati) [inline app/Models/Film.php:11]: `videoNames`

**Review by vistrati — CHANGES_REQUESTED** [review]

File: database/migrations/2021_12_10_190950_payment_table.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::statement("
+            CREATE TABLE `payment` (
+            `id` bigint(20) AUTO_INCREMENT,
```
Comment (egorbwork) [inline database/migrations/2021_12_10_190950_payment_table.php:19]: why not unsigned.

File: database/migrations/2021_12_10_190950_payment_table.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            `id_user` bigint(20) unsigned,
+            `month` int NOT NULL,
+            `date_payment`  DATETIME  NOT NULL,
```
Comment (egorbwork) [inline database/migrations/2021_12_10_190950_payment_table.php:22]: can be just date

File: database/migrations/2021_12_10_214720_genre_table.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::statement("
+        CREATE TABLE `genre`(
+            `id` int NOT NULL  AUTO_INCREMENT,
+            `name` varchar(50) NOT NULL,
```
Comment (egorbwork) [inline database/migrations/2021_12_10_214720_genre_table.php:20]: I recommend unique.

File: database/migrations/2021_12_10_223815_film_table.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::statement("
+        CREATE TABLE `film` (
```
Comment (egorbwork) [inline database/migrations/2021_12_10_223815_film_table.php:18]: Initialy it was named films

File: database/migrations/2021_12_10_224444_pivot_genre_film_table.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::statement("
+        CREATE TABLE `genre_film`(
```
Comment (egorbwork) [inline database/migrations/2021_12_10_224444_pivot_genre_film_table.php:18]: Please fix indentation.

File: database/migrations/2021_12_10_190950_payment_table.php (hunk 3 above)
Comment (egorbwork) [reply database/migrations/2021_12_10_190950_payment_table.php:19]: As helpers were used in few places. You can ignore it.

File: database/migrations/2021_12_10_234843_film_alternative_name_table.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('film_alternative_name', function (Blueprint $table) {
+            $table->id();
+            $table->integer('id_film');
+            $table->string('name', 255);
```
Comment (egorbwork) [inline database/migrations/2021_12_10_234843_film_alternative_name_table.php:20]: Initially it was 50, 255 is planned?

File: database/migrations/2021_12_10_234843_film_alternative_name_table.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            $table->timestamps();
```
Comment (egorbwork) [inline database/migrations/2021_12_10_234843_film_alternative_name_table.php:21]: why do you need timestamps?

File: database/migrations/2021_12_11_001931_film_video_name_table.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('film_video_name', function (Blueprint $table) {
+            $table->id();
+            $table->integer('id_film');
+            $table->string('name', 255);
+            $table->timestamps();
```
Comment (egorbwork) [inline database/migrations/2021_12_11_001931_film_video_name_table.php:21]: why do you need timestamps here?

File: database/migrations/2021_12_11_002430_film_info_table.php [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('film_info', function (Blueprint $table) {
+            $table->id();
+            $table->integer('id_film')->unique();
+            $table->string('lang');
```
Comment (egorbwork) [inline database/migrations/2021_12_11_002430_film_info_table.php:20]: Length is missing for all varchars.

File: database/migrations/2021_12_11_002430_film_info_table.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+            $table->string('screenwriter');
+            $table->string('actors');
+            $table->string('country');
+            $table->integer('duration');
+            $table->timestamps();
```
Comment (egorbwork) [inline database/migrations/2021_12_11_002430_film_info_table.php:25]: why do you need timestamps here?

File: database/migrations/2021_12_11_012611_add_payment_colum_in_payment_table.php [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('payment', function (Blueprint $table) {
+            $table->decimal('payment', $precision = 8, $scale = 2);
```
Comment (egorbwork) [inline database/migrations/2021_12_11_012611_add_payment_colum_in_payment_table.php:17]: I recommend to call this column value.

File: database/migrations/2021_12_11_014853_drop_lang_colum_in_film_info_table.php [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('film_info', function (Blueprint $table) {
+            $table->string('lang');
```
Comment (egorbwork) [inline database/migrations/2021_12_11_014853_drop_lang_colum_in_film_info_table.php:29]: please add length.

**Review by egorbwork — CHANGES_REQUESTED** [review]

File: app/Models/Film.php (hunk 1 above)
Comment (egorbwork) [reply app/Models/Film.php:19]: I once corrected him. But looks as it remained just in table names.

**Comment by egorbwork** [comment]
.gitignore from project was not commited or is empty. It is not acceptable.

**Comment by egorbwork** [comment]
example of .gitignore content:
```
/node_modules
/public/hot
/public/storage
/storage/*.key
/vendor
.env
.env.backup
.phpunit.result.cache
docker-compose.override.yml
Homestead.json
Homestead.yaml
npm-debug.log
yarn-error.log
/.idea
/.vscode
```

**Comment by egorbwork** [comment]
How is fixable:
https://stackoverflow.com/questions/1143796/remove-a-file-from-a-git-repository-without-deleting-it-from-the-local-filesyste

File: app/Models/Film.php (hunk 1 above)
Comment (maiklgravi) [reply app/Models/Film.php]: changed thanks!

File: app/Models/Film.php (hunk 2 above)
Comment (maiklgravi) [reply app/Models/Film.php]: changed thanks!

File: database/migrations/2021_12_10_190950_payment_table.php (hunk 3 above)
Comment (maiklgravi) [reply database/migrations/2021_12_10_190950_payment_table.php:19]: changed thanks!

File: database/migrations/2021_12_10_190950_payment_table.php (hunk 4 above)
Comment (maiklgravi) [reply database/migrations/2021_12_10_190950_payment_table.php:22]: I do this

File: database/migrations/2021_12_10_214720_genre_table.php (hunk 5 above)
Comment (maiklgravi) [reply database/migrations/2021_12_10_214720_genre_table.php:20]: yes i used it

File: database/migrations/2021_12_10_223815_film_table.php (hunk 6 above)
Comment (maiklgravi) [reply database/migrations/2021_12_10_223815_film_table.php:18]: yes my mistake

File: database/migrations/2021_12_11_001931_film_video_name_table.php (hunk 10 above)
Comment (maiklgravi) [reply database/migrations/2021_12_11_001931_film_video_name_table.php:21]: yes not need fhanks

File: database/migrations/2021_12_11_002430_film_info_table.php (hunk 11 above)
Comment (maiklgravi) [reply database/migrations/2021_12_11_002430_film_info_table.php:20]: i do this

File: database/migrations/2021_12_11_002430_film_info_table.php (hunk 12 above)
Comment (maiklgravi) [reply database/migrations/2021_12_11_002430_film_info_table.php:25]: yes not need fhanks

File: database/migrations/2021_12_11_012611_add_payment_colum_in_payment_table.php (hunk 13 above)
Comment (maiklgravi) [reply database/migrations/2021_12_11_012611_add_payment_colum_in_payment_table.php:17]: i do this

File: database/migrations/2021_12_11_014853_drop_lang_colum_in_film_info_table.php (hunk 14 above)
Comment (maiklgravi) [reply database/migrations/2021_12_11_014853_drop_lang_colum_in_film_info_table.php:29]: i do it

File: database/migrations/2021_12_10_224444_pivot_genre_film_table.php (hunk 7 above)
Comment (maiklgravi) [reply database/migrations/2021_12_10_224444_pivot_genre_film_table.php]: i fix that

File: database/migrations/2021_12_11_001931_film_video_name_table.php [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+class FilmVideoNameTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('film_video_name', function (Blueprint $table) {
```
Comment (egorbwork) [inline database/migrations/2021_12_11_001931_film_video_name_table.php:17]: initially table name was without prefix `film` is intended?

File: database/migrations/2021_12_11_012611_add_payment_colum_in_payment_table.php [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('value', function (Blueprint $table) {
+            $table->dropColumn('payment');
```
Comment (egorbwork) [inline database/migrations/2021_12_11_012611_add_payment_colum_in_payment_table.php:29]: The rollback has now invalid column name

**Review by egorbwork — CHANGES_REQUESTED** [review]

File: database/migrations/2021_12_10_214720_genre_table.php [hunk 17]
```diff
@@ -13,7 +14,11 @@ class {{ class }} extends Migration
      */
     public function up()
     {
-        //
+        DB::statement("
+        CREATE TABLE `genre`(
+        `id` int NOT NULL  AUTO_INCREMENT,
+        `name` varchar(50) NOT NULL unique,
+        PRIMARY KEY(id))ENGINE=InnoDB");
```
Comment (egorbwork) [inline database/migrations/2021_12_10_214720_genre_table.php:21]: indentation is not really readable

File: database/migrations/2021_12_10_234843_film_alternative_name_table.php [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+class FilmAlternativeNameTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('film_alternative_name', function (Blueprint $table) {
```
[the remark this answers is not in the document]
Comment (egorbwork) [reply database/migrations/2021_12_10_234843_film_alternative_name_table.php:17]: any response?

File: database/migrations/2021_12_10_234843_film_alternative_name_table.php [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+            $table->id();
+            $table->integer('id_films');
+            $table->string('name', 30);
```
[the remark this answers is not in the document]
Comment (egorbwork) [reply database/migrations/2021_12_10_234843_film_alternative_name_table.php:20]: any response?

File: database/migrations/2021_12_10_224444_pivot_genre_film_table.php [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+        DB::statement("
+        CREATE TABLE `genre_films`(
+        `id_films` int NOT NULL,
+        `id_genre` int NOT NULL,
+        FOREIGN KEY (id_films) REFERENCES films(id)
+        ON UPDATE CASCADE
+        ON DELETE CASCADE,
+        FOREIGN KEY (id_genre) REFERENCES genre(id)
+        ON UPDATE CASCADE
+        ON DELETE CASCADE)ENGINE=InnoDB");
```
[the remark this answers is not in the document]
Comment (egorbwork) [reply database/migrations/2021_12_10_224444_pivot_genre_film_table.php:26]: any response?

File: database/migrations/2021_12_11_001931_film_video_name_table.php (hunk 15 above)
Comment (egorbwork) [reply database/migrations/2021_12_11_001931_film_video_name_table.php:17]: any response?

File: database/migrations/2021_12_11_002430_film_info_table.php [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('film_info', function (Blueprint $table) {
+            $table->id();
+            $table->integer('film_id',15)->unique();
```
Comment (egorbwork) [inline database/migrations/2021_12_11_002430_film_info_table.php:19]: should be just unsigned int.

File: database/migrations/2021_12_11_002430_film_info_table.php [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+            $table->string('lang',15);
+            $table->string('screenwriter',35);
+            $table->string('actors',50);
+            $table->string('country',20);
```
Comment (egorbwork) [inline database/migrations/2021_12_11_002430_film_info_table.php:23]: why length were changed?

**Review by egorbwork — CHANGES_REQUESTED** [review]
Not answered comments that were resolved and some places for fixes. Not acceptable for the moment.

File: database/migrations/2021_12_11_014853_drop_lang_colum_in_film_info_table.php [hunk 23]
```diff
@@ -25,8 +25,8 @@ public function up()
      */
     public function down()
     {
-        Schema::table('{{ table }}', function (Blueprint $table) {
-            //
+        Schema::table('film_info', function (Blueprint $table) {
+            $table->string('lang',15);
```
Comment (egorbwork) [inline database/migrations/2021_12_11_014853_drop_lang_colum_in_film_info_table.php:29]: Rollback is invalid, 50 is used at table creation.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Unfortunately the rollback for one migration is invalid. I won't anymore check this pr, so you are on your own.

**Pull request closed without merging**